### ISSUE TITLE
Saving and loading a model using tf.keras and CategoricalCrossentropy with from_logits=True restores an incorrect model.

### ISSUE THREAD

**Issue by dcbarton** [tensorflow/tensorflow#47475]
Only incorrect when using CategoricalCrossentropy with from_logits=True and sigmoid outputs. Using from_logits=False and softmax outputs is fine, as is BinaryCrossenropy with from_logits=True and sigmoid outputs.

Big problem since CategoricalCrossentropy is most popular loss, and from_logits=True is recommended usage due to numerical stability.

Model is saved using ModelCheckpoint, val_accuracy, max. Trained using Adam.

Saved model loads successfully but predictions do not match the validation accuracy reported by tf when the model was saved, always much worse but does indicate partial training rather than random weights. 

Loaded model object reports correct from_logits value.

The first saved model usually gives correct result, the second save is often is wrong, third and beyond are always wrong.

Even a simple models displays this behavior.

Tested on stable and nightly tf, same result. Using tf.keras, not standalone keras.

**Comment by Saduf2019** [comment]
@dcbarton 
We see that the issue template has not been filled, could you please do so as it helps us analyse the issue [tf version, steps followed before you ran into this error or stand alone code to reproduce the issue faced]

**Comment by dcbarton** [comment]
I'm filing this as a courtesy and gave detailed info. I don't know what template you are referring to. My tf version is:

v1.12.1-51802-gb334cd537ac 2.5.0-dev20210228

I'm on ubuntu 20.04, cuda 11.1, cuDnn 8.1

**Comment by Saduf2019** [comment]
@dcbarton 
Please hare simple stand alone code as requested for us to replicate the issue and analyse or if possible share a colab gist with the error reported.
